e2e/poms/task1.ts

```diff
@@ -0,0 +1,190 @@
+import { expect, Page } from "@playwright/test";
+import {TASK1SELECTOR} from '../constants/selectors/task1Selectors';
+import { TASK1TEXT } from "../constants/texts/task1texts";
+
+export class Task1 {
+    constructor(private page: Page) {}
+
+    async buildPage() {
+        await this.page.goto(TASK1SELECTOR.urls.home);
+        await this.page.getByTestId(TASK1SELECTOR.testIds.addFormButton).click();
+        await this.page.getByTestId(TASK1SELECTOR.testIds.startFromScratchButton).click();
+    }
+    
+    async buildFor1() {
+        await this.page.getByTestId(TASK1SELECTOR.testIds.addFullNameElement).click();
+        await expect(this.page.getByTestId(TASK1SELECTOR.testIds.fullNamePreviewGroup)).toBeVisible();
+        await this.page.getByTestId(TASK1SELECTOR.testIds.addPhoneNumberElement).click();
+        await expect(this.page.getByTestId(TASK1SELECTOR.testIds.phonePreviewGroup)).toBeVisible();
+
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishButton).click();
+        const previewPromise = this.page.waitForEvent("popup");
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishPreviewButton).click();
+        const page2 = await previewPromise;
+        return page2;
+    }
+
+    async submit(page2: Page) {
+        await expect(page2.getByTestId(TASK1SELECTOR.testIds.phoneGroup)).toBeVisible();
+        await expect(page2.getByTestId(TASK1SELECTOR.testIds.fullNameGroup)).toBeVisible();
+        await expect(page2.getByTestId(TASK1SELECTOR.testIds.emailGroup)).toBeVisible();
```

Comment:
When dealing with multiple back-to-back expects, it is better to make it a `Promise.all` block for better performance.

```suggestion
        await Promise.all([
          expect(page2.getByTestId(TASK1SELECTOR.testIds.phoneGroup)).toBeVisible();
          expect(page2.getByTestId(TASK1SELECTOR.testIds.fullNameGroup)).toBeVisible();
          expect(page2.getByTestId(TASK1SELECTOR.testIds.emailGroup)).toBeVisible();
        ])
```